pkg/datapath/loader/netlink.go

```diff
@@ -134,20 +151,29 @@ func replaceDatapath(ctx context.Context, ifName, objPath, progName, direction s
 		return nil, err
 	}
 
-	l.Debug("Attaching program to interface")
-	if err := attachProgram(link, coll.Programs[progName], directionToParent(direction), xdpModeToFlag(xdpMode)); err != nil {
-		// Program replacement unsuccessful, revert bpffs migration.
+	attachedProgsCount := 0
+	for i, prog := range progsToAttach {
+		scopedLog := l.WithField("device", prog.ifName).WithField("progName", prog.progName).
+			WithField("direction", prog.direction).WithField("ifindex", links[i].Attrs().Index)
+		scopedLog.Debug("Attaching program to interface")
+		if err := attachProgram(links[i], coll.Programs[prog.progName], directionToParent(prog.direction), xdpModeToFlag(prog.xdpMode)); err != nil {
+			errs = multierr.Append(errs, fmt.Errorf("program %s: %w", prog.progName, err))
```

Comment:
Same here, this should bail out by reverting the migration and returning `nil, err`.